pkg/config/rules.go

```diff
@@ -0,0 +1,201 @@
+package config
+
+import (
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+)
+
+func (c *Config) AddSignRule(accountName string, rule SignRule) (string, error) {
+	if err := validateRule(rule); err != nil {
+		return "", fmt.Errorf("invalid rule: %v", err)
+	}
+
+	rule.ID = generateRuleID(rule)
+
+	configManager.mu.Lock()
+	defer configManager.mu.Unlock()
+
+	// Check if the account exists
+	account, exists := c.Accounts[accountName]
+	if !exists {
+		account = AccountCfg{SignRules: []SignRule{}}
+		c.Accounts[accountName] = account
+	}
+
+	// Add the new rule
+	account.SignRules = append(account.SignRules, rule)
+	c.Accounts[accountName] = account
+
+	err := saveConfigUnsafe(c)
+	if err != nil {
+		return "", fmt.Errorf("error saving configuration: %v", err)
+	}
+
+	return rule.ID, nil
+}
+
+func (c *Config) DeleteSignRule(accountName, ruleID string) error {
+	configManager.mu.Lock()
+	defer configManager.mu.Unlock()
+
+	// Check if the account exists
+	account, exists := c.Accounts[accountName]
+	if !exists {
+		return fmt.Errorf("account not found: %s", accountName)
+	}
+
+	// Find the index of the rule to delete
+	index := -1
+
+	for i, rule := range account.SignRules {
+		if rule.ID == ruleID {
+			index = i
+			break
+		}
+	}
+
+	if index == -1 {
+		return fmt.Errorf("rule not found: %s", ruleID)
+	}
+
+	// Remove the rule from the slice
+	account.SignRules = append(account.SignRules[:index], account.SignRules[index+1:]...)
+	c.Accounts[accountName] = account
+
+	err := saveConfigUnsafe(c)
+	if err != nil {
+		return fmt.Errorf("error saving configuration: %v", err)
+	}
+
+	return nil
+}
+
+func (c *Config) UpdateSignRule(accountName, ruleID string, newRule SignRule) (string, error) {
+	configManager.mu.Lock()
+	defer configManager.mu.Unlock()
+
+	// Check if the account exists
+	account, exists := c.Accounts[accountName]
+	if !exists {
+		return "", fmt.Errorf("account not found: %s", accountName)
+	}
+
+	// Find the index of the rule to update
+	index := -1
+
+	for i, rule := range account.SignRules {
+		if rule.ID == ruleID {
+			index = i
+			break
+		}
+	}
```

Comment:
Could be nice to have it in another function as it's also used in `DeleteSignRule`